assemblies/assembly-optimizer-migration-8.adoc

```diff
@@ -0,0 +1,1685 @@
+
+[source,xml]
+----
+    <plannerBenchmark>
+      ...
+      <parallelBenchmarkCount>availableProcessorCount - 1</parallelBenchmarkCount>
+      ...
+    </plannerBenchmark>
+----
+
+After in `benchmarkConfig.xml`:
+
+[source,xml]
+----
+    <plannerBenchmark>
+      ...
+      <parallelBenchmarkCount>1</parallelBenchmarkCount> <!-- Alternatively, use "AUTO" or omit entirely. -->
+      ...
+    </plannerBenchmark>
+----
+
+[discrete]
+=== Removed the deprecated variable listeners
+*Major*, *Public API*
+
+A deprecated interface `VariableListener` from package `org.optaplanner.core.impl.domain.variable.listener` has ben removed,
+along with a deprecated interface `StatefulVariableListener`
+and a deprecated class `VariableListenerAdapter` in that same package.
+Instead, use an interface `VariableListener` from package `org.optaplanner.core.api.domain.variable`.
+
+Before in `*VariableListener.java*`:
+
+[source,java]
+----
+  ...
+  import org.optaplanner.core.impl.domain.variable.listener.VariableListenerAdapter;
+  ...
+
+  public class MyVariableListener extends VariableListenerAdapter<Object> {
+
+    ...
+
+    @Override
+    void afterEntityRemoved(ScoreDirector scoreDirector, Object entity);
+      ...
+    }
+
+    ...
+  }
+
+----
+
+After in `*VariableListener.java*`:
+
+[source,java]
+----
+  ...
+  import org.optaplanner.core.api.domain.variable.VariableListener;
+  ...
+
+  public class MyVariableListener extends VariableListener<MySolution, Object> {
+
+    ...
+
+    @Override
+    void afterEntityRemoved(ScoreDirector<MySolution> scoreDirector, Object entity);
+      ...
+    }
+
+    ...
+  }
+----
+
+Before in `*StatefulVariableListener.java*`:
+
+[source,java]
+----
+  ...
+  import org.optaplanner.core.impl.domain.variable.listener.StatefulVariableListener;
+  ...
+
+  public class MyStatefulVariableListener implements StatefulVariableListener<Object> {
+
+    ...
+
+    @Override
+    public void clearWorkingSolution(ScoreDirector scoreDirector) {
+      ...
+    }
+
+    ...
+  }
+
+----
+
+After in `*StatefulVariableListener.java*`:
+
+[source,java]
+----
+  ...
+  import org.optaplanner.core.api.domain.variable.VariableListener;
+  ...
+
+  public class MyStatefulVariableListener implements VariableListener<MySolution, Object> {
+
+    ...
+
+    @Override
+    public void close() {
+      ...
+    }
+
+    ...
+  }
+
+----
+
+== From 8.2.0.Final to 8.3.0.Final
+
+[discrete]
+=== `ConstraintMatch.compareTo()` inconsistent with `equals()`
+*Minor*
+
+The `equals()` override in `ConstraintMatch` has been removed.
+As a result, two different `ConstraintMatch` instances are never considered equal.
+This is in contrast to the `compareTo()` method, which continues to consider two instances equal
+if all their field values are equal.
+
+The `equals()` override in `ConstraintMatch` has been removed
+in order to not fail on constraints with non-distinct matches.
+
+== From 8.5.0.Final to 8.6.0.Final
```

Comment:
This section should be removed as OptaPlanner 8.6.0.Final is not included in the product 7.11 release.